[11.6] Add support for uploading avatars

TecsiAron opened the pull request:
Recently migrated to GitLab from SVN. Most of our projects have a project icon at a specific path in the repo, needed an automatic way of uploading these icons to GitLab.
NOTE works on Gitlab 12.9+

TecsiAron committed:
Add support for uploading avatars

GrahamCampbell reviewed (changes requested):
Please add test coverage for this. Then, I'll be happy to accept.

TecsiAron commented:
Added a test. It is my first time doing any kind of unit testing in PHP or on a REST API, so I have no idea if I did it right. Let me know, thanks.
(When executed successfully the method returns the same as Projects->show() so I took the expectedArray from there, and added the avatar_url, used the base URL from other tests in the same file.)

GrahamCampbell commented:
NB It's not intended that you enable StyleCI on forks. StyleCI can apply the fixes when this PR is merged.

TecsiAron commented:
Cleaned up the commits.
Let me know if any further changes are needed. 
(I ran StyleCI on the fork because I got an error on my original commit stating that the formatting is wrong.)

GrahamCampbell commented:
> I ran StyleCI on the fork because I got an error on my original commit stating that the formatting is wrong.

It is still not necessary to do this.

GrahamCampbell commented:
Thanks for the PR.
